<?xml version="1.0" ?>
<sdf version="1.5">
  <world name="default">

    <physics type="ode">
      <max_step_size>0.01</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>100</real_time_update_rate>
    </physics>


    <plugin name="hunav_plugin" filename="libHuNavPlugin.so">
      <update_rate>100.0</update_rate>
      <robot_name>actor3</robot_name>
      <use_gazebo_obs>True</use_gazebo_obs>
      <global_frame_to_publish>map</global_frame_to_publish>
      <ignore_models>
        <model>cafe</model>
        <model>ground_plane</model>
      </ignore_models>
    </plugin>
    
    <gui>
      <camera name="user_camera">
        <pose>0 -18 8.0 0 0.523 1.5707</pose>
      </camera>
    </gui>
    <light name="sun" type="directional">
      <cast_shadows>1</cast_shadows>
      <pose>0 0 10 0 -0 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>0 0.5 -0.9</direction>
    </light>

    <light name="point_light" type="point">
      <pose>0 -0.377195 6.59868 0 -0 0</pose>
      <diffuse>0.784314 0.784314 0.784314 1</diffuse>
      <specular>0.0980392 0.0980392 0.0980392 1</specular>
      <attenuation>
        <range>50</range>
        <constant>0.8</constant>
        <linear>0</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <cast_shadows>0</cast_shadows>
      <direction>0 0 -1</direction>
    </light>

    <include>
      <uri>model://ground_plane</uri>
    </include>

    <include>
      <uri>model://cafe</uri>
    </include>

    <include>
      <name>table1</name>
      <pose>0.5 -1.6 0.2 0 0 0</pose>
      <uri>model://cafe_table</uri>
    </include>

    <include>
      <name>table2</name>
      <pose>2.4 -5.5 0.2 0 0 0</pose>
      <uri>model://cafe_table</uri>
    </include>

    <include>
      <!--<static>false</static>-->
      <name>table3</name>
      <pose>-1.5 -5.5 0.2 0 0 0</pose>
      <uri>model://cafe_table</uri>
    </include>

    <include>
      <name>table4</name>
      <pose>2.4 -9 0.2 0 0 0</pose>
      <uri>model://cafe_table</uri>
    </include>

    <include>
      <name>table5</name>
      <pose>-1.5 -9 0.2 0 0 0</pose>
      <uri>model://cafe_table</uri>
    </include>

    <actor name="actor1-green"> 
      <pose>-1 2 1.25 0 0 0</pose>
      <skin>
        <filename>walk-green.dae</filename>
        <scale>1.0</scale>
      </skin>
      <animation name="no_active">
        <filename>07_01-walk.bvh</filename>
        <scale>1.000000</scale>
        <interpolate_x>true</interpolate_x>
      </animation>
      <animation name="active">
        <filename>17_01-walk_with_anger.bvh</filename>
        <scale>1.000000</scale>
        <interpolate_x>true</interpolate_x>
      </animation>
    </actor>

    <actor name="actor3">
      <pose>-1 0 1.25 0 0 0</pose>
      <skin>
        <filename>stand.dae</filename>
        <scale>1.0</scale>
      </skin>
      <!--<animation name="stand">
        <filename>141_20-waiting.bvh</filename>
        <scale>1.000000</scale>
        <interpolate_x>true</interpolate_x>
      </animation>-->
    </actor>

 
    <actor name="actor2-blue">
      <pose>-1 -1 1.25 0 0 0</pose>
      <skin>
        <filename>walk-blue.dae</filename> 
        <scale>1.0</scale>
      </skin>
      <animation name="no_active">
        <filename>07_01-walk.bvh</filename>
        <scale>1.0</scale>
        <interpolate_x>true</interpolate_x>
      </animation>
      <animation name="active">
        <filename>142_17-walk_scared.bvh</filename>
        <scale>1.0</scale>
        <interpolate_x>true</interpolate_x>
      </animation>
    </actor>

	

    <actor name="actor4-red">
      <pose>3 0 1.25 0 0 1.57</pose>
      <skin>
        <filename>walk-red.dae</filename> 
        <scale>1.0</scale>
      </skin>
      <animation name="no_active">
        <filename>142_04-walk_cool.bvh</filename>
        <scale>1.000000</scale>
        <interpolate_x>true</interpolate_x>
      </animation>
      <animation name="active">
        <filename>120_19-walk_slow.bvh</filename>
        <scale>1.0</scale>
        <interpolate_x>true</interpolate_x>
      </animation>
    </actor>


    <!--<actor name="actor5">
      <pose>3 0.8 1.25 0 0 0</pose>
      <skin>
        <filename>walk.dae</filename> 
        <scale>1.0</scale>
      </skin>
      <animation name="walking">
        <filename>walk.dae</filename>
        <scale>1.000000</scale>
        <interpolate_x>true</interpolate_x>
      </animation>
    </actor>-->

  </world>
</sdf>
